src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -226,6 +239,57 @@ public Mono<VposAuthResponseDto> requestCreditCardAuthorization(AuthorizationReq
                 );
     }
 
+    public Mono<StateResponseDto> requestNpgCardsAuthorization(AuthorizationRequestData authorizationData) {
+        return Mono.just(authorizationData)
+                .filter(
+                        authorizationRequestData -> "CP".equals(authorizationRequestData.paymentTypeCode())
+                                && NpgClient.PaymentMethod.CARDS.serviceName
+                                        .equals(authorizationRequestData.paymentMethodName())
+                )
+                .switchIfEmpty(Mono.empty())
+                .filter(
+                        authorizationRequestData -> authorizationData
+                                .authDetails() instanceof CardsAuthRequestDetailsDto
+                )
+                .switchIfEmpty(
+                        Mono.error(
+                                new InvalidRequestException(
+                                        "Cannot perform NPG authorization for invalid input CardsAuthRequestDetailsDto"
+                                )
+                        )
+                )
+                .flatMap(authorizationRequestData -> {
+                    CardsAuthRequestDetailsDto cardsData = (CardsAuthRequestDetailsDto) authorizationData.authDetails();
+                    final BigDecimal grandTotal = BigDecimal.valueOf(
```

Comment:
```suggestion
                .map(authorizationRequestData -> authorizationData.authDetails())
                .cast(CardsAuthRequestDetailsDto.class)
                .flatMap(cardsData -> {
                    final BigDecimal grandTotal = BigDecimal.valueOf(
```
Maybe with this approach here can avoid explicit casting. What do you think?

Reply:
good tip :-)